### PR TITLE
[AppService] Add support for deployment logs for webapp in CLI

### LINKED ISSUE #13734: az webapp logs needs to support deployment logs
We can get list of deployments form: 

`https://<webapp>.scm.azurewebsites.net/api/deployments/`

And you can get deployment logs form a specific deployment through:

`https://<webapp>.scm.azurewebsites.net/api/deployments/<id>`

### PR DESCRIPTION
**Pull request by calvinsID**
**Description** 

closes #13734 
Add support for new webapp commands for deployment logs: `az webapp log deployment show` and `az webapp log deployment list`. 

**Testing Guide**  

Create & deploy a web or function app

Run `az webapp log deployment list -g ResourceGroup -n WebappName`. This should give you the same list of deployment logs you see when you go to `https://<webapp>.scm.azurewebsites.net/api/deployments`

Run `az webapp log deployment show -g ResourceGroup -n WebappName`. This should give you the latest deployment log

Copy the Deployment ID from any deployment

Run `az webapp log deployment show -g ResourceGroup -n WebappName --deployment-id DeploymentId`. This should give you the log of that particular deployment.

**History Notes**  

[App Service] `az webapp log deployment show`: Show the latest deployment log, or the deployment logs of a specific deployment if deployment-id is specified
[App Service] `az webapp log deployment list`: List of deployment logs available

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S171

**Comment by calvinsID** [comment]
@btardif Thoughts on the command names? Changed from `logs deployment` -> `log deployment` because that group existed already. Added `show` for viewing a single deployment log

File: src/azure-cli/azure/cli/command_modules/appservice/_help.py [hunk 1]
```diff
@@ -1668,6 +1668,52 @@
 long-summary: This command may not work with web apps running on Linux.
 """
 
+helps['webapp log deployment'] = """
+type: group
+short-summary: Manage web app deployment logs.
+"""
+
+helps['webapp log deployment show'] = """
```
Comment (panchagnula) [inline src/azure-cli/azure/cli/command_modules/appservice/_help.py:1676]: @btardif can you please review the help content & let us know if we need to update these? Thanks!

File: src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_webapp_commands.py [hunk 2]
```diff
@@ -2400,6 +2400,58 @@ def test_webapp_vnetDelegation(self, resource_group):
         ])
 
 
+class WebappDeploymentLogsScenarioTest(LocalContextScenarioTest):
+    @ResourceGroupPreparer()
+    def test_webapp_show_deployment_logs(self, resource_group):
```
Comment (panchagnula) [inline src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_webapp_commands.py:2405]: @calvinsID this doesn't test for functionapp can we update an existing function app E2E test to test show deployment

File: src/azure-cli/azure/cli/command_modules/appservice/commands.py [hunk 3]
```diff
@@ -185,6 +185,14 @@ def load_command_table(self, _):
         g.custom_command('config', 'config_diagnostics', validator=validate_app_or_slot_exists_in_rg)
         g.custom_show_command('show', 'show_diagnostic_settings', validator=validate_app_or_slot_exists_in_rg)
 
+    with self.command_group('webapp log deployment') as g:
+        g.custom_command('show', 'show_deployment_log')
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/appservice/commands.py:189]: Maybe `g.custom_show_command` is better

File: src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_webapp_commands.py (hunk 2 above)
Comment (calvinsID) [reply src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_webapp_commands.py:2472]: Added!

**Pull request merged**